tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_DampedHarmonicGaugeQuantities.cpp

```diff
@@ -0,0 +1,132 @@
+  pypp::check_with_random_values<1>(
+      static_cast<double (*)(const double&, const double&, const double&)>(
+        &::GeneralizedHarmonic::roll_on_function),
+      "TestFunctions", "roll_on_func",
+      {{{std::numeric_limits<double>::denorm_min(), 10.0}}}, used_for_size);
+  // time_deriv_of_roll_on_function
+  pypp::check_with_random_values<1>(
+      static_cast<double (*)(const double&, const double&, const double&)>(
+        &::GeneralizedHarmonic::time_deriv_of_roll_on_function),
+      "TestFunctions", "time_deriv_roll_on_func",
+      {{{std::numeric_limits<double>::denorm_min(), 10.0}}}, used_for_size);
+}
+
+template <size_t SpatialDim, typename Frame>
+void test_damped_harmonic_h_function(const DataVector& used_for_size) noexcept {
+  // H_a
+  pypp::check_with_random_values<1>(
+      static_cast<typename GeneralizedHarmonic::GaugeH<SpatialDim,
+          Frame>::type (*)(
+        const typename
+          ::GeneralizedHarmonic::InitialGaugeH<SpatialDim, Frame>::type&,
+        const Scalar<DataVector>&,
+        const tnsr::I<DataVector, SpatialDim, Frame>&,
+        const Scalar<DataVector>&,
+        const tnsr::aa<DataVector, SpatialDim, Frame>&,
+        const double&, const double&, const double&,
+        const tnsr::I<DataVector, SpatialDim, Frame>&,
+        const double&, const double&)>(
+          &::GeneralizedHarmonic::DampedHarmonicHCompute<SpatialDim,
+          Frame>::function), "TestFunctions", "gauge_h_func",
+      {{{std::numeric_limits<double>::denorm_min(), 10.0}}}, used_for_size);
+}
+
+template <size_t SpatialDim, typename Frame>
+void test_deriv_damped_harmonic_h_function(const DataVector& used_for_size)
+  noexcept {
+  pypp::check_with_random_values<1>(
+      static_cast<typename GeneralizedHarmonic::SpacetimeDerivGaugeH<SpatialDim,
+          Frame>::type (*)(
+        const typename
+          ::GeneralizedHarmonic::InitialGaugeH<SpatialDim, Frame>::type&,
+        const typename ::GeneralizedHarmonic::SpacetimeDerivInitialGaugeH<
+          SpatialDim, Frame>::type&,
+        const Scalar<DataVector>&,
+        const tnsr::I<DataVector, SpatialDim, Frame>&,
+        const tnsr::A<DataVector, SpatialDim, Frame>&,
+        const tnsr::a<DataVector, SpatialDim, Frame>&,
+        const Scalar<DataVector>&,
+        const tnsr::II<DataVector, SpatialDim, Frame>&,
+        const tnsr::aa<DataVector, SpatialDim, Frame>&,
+        const tnsr::iaa<DataVector, SpatialDim, Frame>&,
+        const tnsr::aa<DataVector, SpatialDim, Frame>&,
+        const tnsr::iaa<DataVector, SpatialDim, Frame>&,
+        const Scalar<DataVector>&,
+        const double&, const double&, const double&,
+        const tnsr::I<DataVector, SpatialDim, Frame>&,
+        const double&, const double&)>(
+        &::GeneralizedHarmonic::SpacetimeDerivDampedHarmonicHCompute<SpatialDim,
+          Frame>::function), "TestFunctions", "spacetime_deriv_gauge_h_func",
+      {{{0.1, 10.0}}}, used_for_size);
+}
+
```

Comment:
I'm impressed how fast you got this coded, @prayush! Nicely done. 

Can you please test this on Kerr, or at least Schwarzschild, besides the fuzzy tests? This is complicated enough that I worry it's easy to make the same mistake in c++ and python.

Reply:
@geoffrey4444 - KerrSchild tests are now ready for your perusal

Reply:
comment by @hrueter and myself:

We agree with @geoffrey4444 that, in addition to the comparison to your Python implementation, this needs one (or a few) tests against (ideally) an analytic expression. Right now the KerrSchild test you have implemented below does not do that, since the code to compute $H$ is effectively copied from the C++ source that is being tested. So it currently does not add much (or any) value to the previous tests. So has anyone written down the analytic expression for the damped harmonic $H$ for Kerr in KerrSchild coords, for instance? Or even simpler spacetime+coords, like Schwarzschild/Minkowski in some coords like isotropic/Kerr-Schild/Finkelstein? You can first do a test without the roll-on and weight functions, then test those in another test by computing $H$ at some time during/after roll-on (with some initial $H$, like $H=0$ to start with). Clearly these tests are not the easiest to work out, but we think it's worth it. We can discuss in a hangout if you like.

Reply:
Hi @nilsleiffischer @hrueter - I am happy to add more tests, but I'd be happier if I could pose those as a separate PR! Can we discuss this in a hangout please?

Reply:
As we had discussed, I have reduced analytic tests for Kerr spacetime in Kerr-Schild coordinates into smaller unit tests that now test individual terms in the gauge source function.

Reply:
@geoffrey4444 , @nilsleiffischer , @hrueter : I added some 'analytic' tests for both gauge source function H and its spacetime derivatives for Schwarzschild spacetime. In these, I check for the equality of the output from DampedHarmonicHCompute and SpacetimeDerivDampedHarmonicHCompute ComputeTag's, against ab-initio construction of both quantities for Schwarzschild spacetime in Kerr-Schild coordinates. The choice of coordinates is arbitrary, so I thought KS are as good as any.